pkg/platform/provider/baremetal/cluster/update.go

```diff
@@ -193,6 +194,22 @@ func updateCoreDNSVersion(ctx context.Context, client kubernetes.Interface, vers
 	return err
 }
 
+func (p *Provider) EnsureUpgradeKubeVendor(ctx context.Context, c *v1.Cluster) error {
+	switch c.Spec.KubeVendor {
+	case platformv1.KubeVendorTKE:
+		if vendor.GetKubeVendor(c.Spec.Version) != platformv1.KubeVendorTKE {
+			return fmt.Errorf("not support upgrade from vendor %v to vendor %v", c.Spec.KubeVendor, vendor.GetKubeVendor(c.Spec.Version))
+		}
+	case platformv1.KubeVendorOther:
+		if vendor.GetKubeVendor(c.Spec.Version) != platformv1.KubeVendorOther && vendor.GetKubeVendor(c.Spec.Version) != platformv1.KubeVendorTKE {
+			return fmt.Errorf("not support upgrade from vendor %v to vendor %v", c.Spec.KubeVendor, vendor.GetKubeVendor(c.Spec.Version))
+		}
+	default:
+		return fmt.Errorf("not support upgrade from vendor %v to vendor %v", c.Spec.KubeVendor, vendor.GetKubeVendor(c.Spec.Version))
+	}
+	return nil
+}
+
```

Comment:
1. add KubeVendor to cluster.spec pass by user or UI
2. add KubeVendor  to cluster.status by health-check in controller  
3. make sure this case as same as  `cluster.spac.version` case 
4. if cluster.spec.KubeVendor != cluster.status.KubeVendor  do reconcile in controller even if failed, because this is due to user  invalid input, but we should keep consistent for `controller  reconcile` logic no special handle only for `KubeVendor`
5. that means you should pass KubeVendor to cluster.spec by default when create cluster, user is able to update it during cluster life cycle as long as your  `controller  reconcile` logic handle it correctly